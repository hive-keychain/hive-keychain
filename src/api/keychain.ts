import axios from 'axios';

const KeychainApi = axios.create({
  baseURL:
    process.env.KEYCHAIN_API_DEV === 'true'
      ? 'http://localhost:3000/'
      : 'https://api.hive-keychain.com/',
});

export default KeychainApi;
