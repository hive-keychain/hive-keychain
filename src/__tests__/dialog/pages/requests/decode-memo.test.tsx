import { KeychainKeyTypes } from '@interfaces/keychain.interface';
import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import DecodeMemo from 'src/dialog/pages/requests/decode-memo';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestDecodeMemo from 'src/__tests__/utils-for-testing/data/props/dialog/request-decodeMemo';
import objects from 'src/__tests__/utils-for-testing/helpers/objects';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestDecodeMemo } from 'src/__tests__/utils-for-testing/types/props-types';
describe('decode-memo tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestDecodeMemo;
  methods.config();
  it('Must show DecodeMemo dialog as canWhiteList', async () => {
    const clonedProps = objects.clone(props) as PropsRequestDecodeMemo;
    clonedProps.data.method = KeychainKeyTypes.posting;
    const { asFragment } = render(<DecodeMemo {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot('Request DecodeMemo canWhiteList');
  });
  it('Must show DecodeMemo dialog without can white list', async () => {
    const clonedProps = objects.clone(props) as PropsRequestDecodeMemo;
    clonedProps.data.method = KeychainKeyTypes.active;
    const { asFragment } = render(<DecodeMemo {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot('Request DecodeMemo no whitelist');
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<DecodeMemo {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    render(<DecodeMemo {...{ ...props, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    render(<DecodeMemo {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
