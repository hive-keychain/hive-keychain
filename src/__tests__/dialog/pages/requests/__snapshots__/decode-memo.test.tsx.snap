// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode-memo tests:
 Must show DecodeMemo dialog as canWhiteList: Request DecodeMemo canWhiteList 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Verify Key
      </h2>
      <div
        aria-label="operation_header"
        class="operation_header "
      >
        The website domain would like to verify that you have access to the private Posting key for the account: @keychain.tests
      </div>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Key
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        Posting
      </div>
    </div>
    <div
      class="operation_footer "
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      >
        <div
          class="checkbox-container align-center"
        >
          <div
            class="checkbox-and-label"
          >
            <div
              class="pretty p-default"
            >
              <input
                aria-checked="false"
                id="checkbox-set-keep-inner-input"
                type="checkbox"
                value=""
              />
              <div
                class="state"
              >
                <label
                  for="checkbox-set-keep-inner-input"
                />
              </div>
            </div>
            <div
              aria-label="checkbox-set-keep"
              class="label"
            >
              Do not prompt again to send decode transactions from the @keychain.tests account on domain
            </div>
          </div>
        </div>
      </div>
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`decode-memo tests:
 Must show DecodeMemo dialog without can white list: Request DecodeMemo no whitelist 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Verify Key
      </h2>
      <div
        aria-label="operation_header"
        class="operation_header "
      >
        The website domain would like to verify that you have access to the private Active key for the account: @keychain.tests
      </div>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Key
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        Active
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
